sdk/attestation/azure-security-attestation/samples/policy-certificates/get_policy_certificates.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief This sample demonstrates using the Attestation Service SDK for C++ to retrieve policy
+ * management certificates from an Isolated mode attestation service instance.
+ *
+ * Remember that when an attestation service instance is in isolated mode, the service is configured
+ * with a set of X.509 certificates. The GetPolicyManagementCertificates API returns a list of the
+ * existing certificates.
+ *
+ * @remark The following environment variables must be set before running the sample.
+ * - ATTESTATION_ISOLATED_URL:  Points to an Attestation Service Instance in Isolated mode.
+ * operations.
+ * - AZURE_TENANT_ID:     Tenant ID for the Azure account.
+ * - AZURE_CLIENT_ID:     The Client ID to authenticate the request.
+ * - AZURE_CLIENT_SECRET: The client secret.
+ *
+ */
+
+#include "get_env.hpp"
+
+#include "cryptohelpers.hpp"
+#include <azure/attestation.hpp>
+#include <azure/identity.hpp>
+
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+#include <thread>
+#include <vector>
+
+// cspell:: words mrsigner mrenclave mitm
+using namespace Azure::Security::Attestation;
+using namespace Azure::Security::Attestation::Models;
+using namespace std::chrono_literals;
+using namespace Azure::Core;
+
+int main()
+{
+  try
+  {
+    // create an administration client
+    auto credential = std::make_shared<Azure::Identity::ClientSecretCredential>(
+        std::getenv("AZURE_TENANT_ID"),
+        std::getenv("AZURE_CLIENT_ID"),
+        std::getenv("AZURE_CLIENT_SECRET"));
+    AttestationAdministrationClient adminClient(
+        std::getenv("ATTESTATION_ISOLATED_URL"), credential);
+
+    // Retrieve attestation response validation collateral before calling into the service.
+    adminClient.RetrieveResponseValidationCollateral();
+
+    // Retrieve the SGX Attestation Policy from this attestation service instance.
+    Azure::Response<AttestationToken<PolicyCertificateListResult>> policyCertificates
+        = adminClient.GetPolicyManagementCertificates();
+
+    std::cout << "There are " << policyCertificates.Value.Body.Certificates.size()
+              << " certificates configured on this instance." << std::endl;
+
+    std::cout << "Enumerating policy certificates:" << std::endl;
+    for (const auto& certChain : policyCertificates.Value.Body.Certificates)
+    {
+      auto x509Cert(::Cryptography::ImportX509Certificate((*certChain.CertificateChain)[0]));
+      std::cout << "Subject of signing certificate is: " << x509Cert->GetSubjectName() << std::endl;
+      std::cout << "Issuer of signing certificate is: " << x509Cert->GetIssuerName() << std::endl;
```

Comment:
You use `x509Cert->Foo()` here, but `*certChain.Bar` above.  Consider using the same approach. So maybe change above to `certChain->CertificateChain[0]

Reply:
I'm confused. the `*certChain.CertificateChain` is because `CertificateChain` is an `Azure::Nullable`. `x509Cert` is a std::unique_ptr. They are two different structures with two different access patterns. `certChain->CertificateChain[0]` won't compile because (a) `certChain` is an `AttestationSigner&`, and (b) because `Azure::Nullable` doesn't have an operator[] method.  

Would you prefer that I change it to `certChain.CertificateChain.Value()[0]` instead of `(*certChain.CertificateChain)[0]`?

Reply:
oh! I missed it is a Nullable<T>.
Someone said (don't remember who) in the past that adding `*` and `->` operators to Nullable<T> would eventually bring confusion while reading the code, as you don't know if it is nullable or a raw pointer.  Well, I guess the day's came :)

You might be able to improve readability by either adding a comment or by replacing the `auto` for the Nullable<T>.

Also, if `CertificateChain` is an array, and `importCertificate` takes the type of that array, do we need to pass the index?  Would it by default take the possition 0 ?

Reply:
`ImportCertificate` takes a std::string which represents a single PEM encoded X.509 certificate. `CertificateChain` is an array of PEM encoded X.509 certificates. And I *really* don't want to have to include the OpenSSL logic to validate X.509 certificate chains (especially for a sample).